exercises/sgf-parsing/canonical-data.json

```diff
@@ -0,0 +1,161 @@
+            }
+        },
+        {
+            "description": "properties without delimiter",
+            "property": "parse",
+            "input": {
+                "encoded": "(;a)"
+            },
+            "expected": {
+                "error": "properties without delimiter"
+            }
+        },
+        {
+            "description": "all lowercase property",
+            "property": "parse",
+            "input": {
+                "encoded": "(;a[b])"
+            },
+            "expected": {
+                "error": "property must be all uppercase"
+            }
+        },
+        {
+            "description": "upper and lowercase property",
+            "property": "parse",
+            "input": {
+                "encoded": "(;Aa[b])"
+            },
+            "expected": {
+                "error": "property must be all uppercase"
+            }
+        },
+        {
+            "description": "two nodes",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[B];B[C])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["B"]
+                },
+                "children": [
+                    {
+                        "properties": {
+                            "B": ["C"]
+                        },
+                        "children": []
+                    }
+                ]
+            }
+        },
+        {
+            "description": "two child trees",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[B](;B[C])(;C[D]))"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["B"]
+                },
+                "children": [
+                    {
+                        "properties": {
+                            "B": ["C"]
+                        },
+                        "children": []
+                    },
+                    {
+                        "properties": {
+                            "C": ["D"]
+                        },
+                        "children": []
+                    }
+                ]
+            }
+        },
+        {
+            "description": "multiple property values",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[b][c][d])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["b", "c", "d"]
+                },
+                "children": []
+            }
+        },
+        {
+            "description": "escaped property",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[\\]b\\nc\\nd\\t\\te \\n\\]])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["]b\\nc\\nd  e \\n]"]
+                },
+                "children": []
+            }
+        }
+    ]
+}
```

Comment:
@cmccandless this looks OK to me with the minor exception of the very last line.  

I have not attempted this exercise and so don't have a solid grasp of it in order to form any meaningful opinions one way or the other.  Thanks for undertaking this, it needed to be done. :+1: